Description: Use relative paths for displaying code positions
 Otherwise we end up with the build path in the documentation, making
 the package non-reproducible.
Author: Doug Torrance <dtorrance@piedmont.edu>
Last-Update: 2020-06-02

--- a/M2/Macaulay2/d/actors4.d
+++ b/M2/Macaulay2/d/actors4.d
@@ -1497,7 +1497,7 @@
      locate1();
      p := codePosition(c);
      Expr(Sequence(
-	       toExpr(verifyMinimizeFilename(locatedCode.filename)),
+	       toExpr(relativizeFilename(locatedCode.filename)),
 	       toExpr(locatedCode.minline),
 	       toExpr(locatedCode.mincol),
 	       toExpr(locatedCode.maxline),
@@ -1516,7 +1516,7 @@
 	  then nullE
 	  else Expr(
 	       Sequence(
-		    toExpr(verifyMinimizeFilename(p.filename)),
+		    toExpr(relativizeFilename(p.filename)),
 		    toExpr(int(p.line)),toExpr(int(p.column)),
 		    toExpr(int(p.line)),toExpr(int(p.column)+length(s.symbol.word.name)),
 		    toExpr(int(p.line)),toExpr(int(p.column))
--- a/M2/Macaulay2/d/stdiop.d
+++ b/M2/Macaulay2/d/stdiop.d
@@ -111,7 +111,7 @@
 export tostring(w:Position) : string := (
      if w == dummyPosition 
      then "-*dummy position*-"
-     else errfmt(verifyMinimizeFilename(w.filename),int(w.line),int(w.column + 1),int(w.loadDepth)));
+     else errfmt(relativizeFilename(w.filename),int(w.line),int(w.column + 1),int(w.loadDepth)));
 export (o:file) << (w:Position) : file := o << tostring(w);
 export (o:BasicFile) << (w:Position) : BasicFile := o << tostring(w);
 threadLocal export SuppressErrors := false;
